### ISSUE TITLE
Display packages by category in installer

### ISSUE THREAD

**Issue by jstrosch** [mandiant/flare-vm#578]
### Details

Hey all! Have you considered creating a series of custom configurations that could align to "profiles"? For example, a profile that installs just the tools needed for PE file analysis or malicious documents? This could help reduce install time by helping folks avoid installing packages they don't need. I realize there is a way to customize the install, but these profiles can be passed via command line (XML) to help with anyone that wants/needs more automation (or doesn't want to have to manually tweak the install each time). I've created a few as examples at: https://github.com/jstrosch/FLARE-VM-configs/tree/main. Curious to get everyone's thoughts?

**Comment by Ana06** [comment]
@emtuls recently shared a similar idea with me. I also discussed it with @MalwareMechanic some time ago. But it seems we never created an issue and we didn't have time to work on it. Thanks for documenting the idea @jstrosch!

**Comment by heap-s** [comment]
@Ana06 Is this something that is being considered still?

**Comment by emtuls** [comment]
@heap-s Definitely still being considered though it is unfortunately not on our high priority list at the moment.

We do have a plan to work on improving the GUI interface, primarily adding categories and drop down lists to the package list, and if time permits, we would try to squeeze that in as well. :)

**Comment by Ana06** [comment]
I think organizing tool by categories in the installer UI and allowing to select/unselect all of them would be a nice improvement that addresses this issue and https://github.com/mandiant/flare-vm/issues/432. My understanding is that @emtuls is planing to implement this after implementing https://github.com/mandiant/VM-Packages/issues/1107 as we need a good way to access the categories in the installer. @emtuls I'll assign you the issues to ensure others are aware you are working on it.

**Comment by necrose99** [comment]
winget ui / [unigetui ](https://github.com/marticliment/UniGetUI)  one can make a json apps install  to get custom pkg adds choco / winget pypi scoop etc..

**Comment by Ana06** [comment]
@necrose99 can you elaborate on how that would exactly work? Can we customize the UI with a FLARE-VM theme and the packages to use our feed?

**Comment by Ana06** [comment]
@sara-rn is planing to modify the installer to use https://www.myget.org/F/vm-packages/Packages after she adds the category as [tag in the nuspec](https://learn.microsoft.com/en-us/nuget/reference/nuspec#tags) to display the packages by category and to allow to select a whole category at once. @sara-rn is researching the best way to modify the installer to implement this.

**Comment by Ana06** [comment]
@jstrosch initial idea was a bit different, but raised the concern that we can make a better job helping the user selecting packages. This is what we are trying to address now in this issue by aiming to display the packages by category and selecting a whole category.

@jstrosch note it is already possible to use the config files you shared in your initial comment with `-customConfig https://github.com/jstrosch/FLARE-VM-configs/blob/main/pe-config.xml  -customLayout https://github.com/jstrosch/FLARE-VM-configs/blob/main/pe-CustomStartLayout.xml`

**Comment by necrose99** [comment]
@Ana06  the json feed for winget ui / [unigetui](https://github.com/marticliment/UniGetUI)
you can set in gui , the tool in question , dose [winget](winget.run) chocolatey , node.js , pypi,   scoop  repos..

ie if you install  Uniget , you can get all installed on device , and export a json list. via its gui.. , winget or other repos can upgrade the system
i have a list in json to speed up my home/work/and flare-vm pc's on redeployment  if stuff goes tits up.. 

for prototyping pypi package updates against  flare-vm tools its fast and shareable , as well the more public choco meta/or .install pkgs visual studio 2022 , or also say cutter 

https://community.chocolatey.org/packages/cutter 
https://community.chocolatey.org/packages/rizin

or testing apps  and links ..  i can define my install as one json file and if i share, updates are available , then you'll get them via winget ui / chocolatey etc.

**Comment by sara-rn** [comment]
I encountered several issues during the implementation. The first one was the usage of chocolatey commands, the `-by-tag-only` switch would search across the tags only therefore would have been perfect to match the tags and not the package name. However I ran into errors when using it, see below:
`choco list Hex editors --by-tag-only` worked fine (on a local repo) but when provided a remote repository with `--source/-s https://www.myget.org/F/vm-packages/api/v2` then I would get the following error `When using the '--source' option with the 'choco list' command, only a named alternative source can be provided.`
I found this bug https://github.com/chocolatey/choco/issues/2545, that might be related.
The commands `choco find` and `choco search`, despite having the same switch `--by-tag-only`, wouldn't display any results.
The next option was parsing directly the XML from https://www.myget.org/F/vm-packages/Packages, this option works fine excepts for a limit in the number of packages displayed, if a filter is applied for example https://www.myget.org/F/vm-packages/api/v2/Packages?$filter=IsLatestVersion%20eq%20true it would display more packages, but not all. There must be a way to display all with the API, however this option has been discarded for the moment.

**Comment by Ana06** [comment]
@sara-rn thanks for the research and testing the options to implement the display of packages by categories in the installer, and for documenting the progress.

- `choco list` is not an option as it only supports a named alternative source. This is expected and documented in https://docs.chocolatey.org/en-us/choco/commands/list/:
  
  >  -s, --source=VALUE
  >      Source - Name of alternative source to use, for example 
  >        'windowsfeatures', 'ruby', 'cygwin', or 'python'.

- `choco search` and `choco find` do support a source with the `--by-tag-only` argument. I think this is a chocolatey bug and I think we should report it. I'll open an issue. @sara-rn I recommend you report issues to the open source projects we rely on when you encountered. This collaborative approach helps those projects improve which directly benefits us and I think the projects will appreciate it.

- I have tried several options and I also not able to get (at least the name, description and tags) of ALL packages using the API with something like https://www.myget.org/F/vm-packages/api/v2/Packages?$filter=IsLatestVersion%20eq%20true. Not sure if this is possible.

- Although I think having to install nuspec is not ideal, If we do not have a better alternative I think installing it via `install-packageprovider nuget -force` and then using [Find-Package](https://learn.microsoft.com/en-us/powershell/module/packagemanagement/find-package?view=powershellget-2.x).

**Comment by Ana06** [comment]
@pauby I do not think so. We use meta packages in VM-Packages. But here we want to allow users to select the packages they want to install by displaying all the packages by categories. The users should be able to select only some packages in a category.

**Comment by Ana06** [comment]
@sara-rn has figured out that there is a `<link rel="next"` in `https://www.myget.org/F/vm-packages/api/v2` which we can use to iterate the XML and get all the packages per categories. But the code has turned out to be more complicated than expected. I think that the solution with [Find-Package](https://learn.microsoft.com/en-us/powershell/module/packagemanagement/find-package?view=powershellget-2.x) + installing `nuget` is a better alternative. The code is much simpler making it easier to understand, modify and enhance.

**Comment by Ana06** [comment]
@sara-rn has figured out that there is a `<link rel="next"` in `https://www.myget.org/F/vm-packages/api/v2` which we can use to iterate the XML and get all the packages per categories. It is more complicated than the solution with [Find-Package](https://learn.microsoft.com/en-us/powershell/module/packagemanagement/find-package?view=powershellget-2.x) + installing `nuget`. @sara-rn will finish the implementation in any of the two ways and she will open an issue to consider improving it.
